tests/components/top_bar.test.ts

```diff
@@ -298,50 +298,146 @@ describe("TopBar component", () => {
+      app.destroy();
+    });
+    test.each([
+      ["text", {}, "align-left"],
+      ["0", {}, "align-right"],
+      ["0", { align: "left" }, "align-left"],
+      ["0", { align: "center" }, "align-center"],
+      ["0", { align: "right" }, "align-right"],
+    ])(
+      "alignment icon in top bar match the selected cell",
+      async (content, style, expectedIconClass) => {
+        const model = new Model();
+        setCellContent(model, "A1", content);
+        model.dispatch("SET_FORMATTING", {
+          sheetId: model.getters.getActiveSheetId(),
+          target: [toZone("A1")],
+          style: style as Style,
+        });
+        const { app } = await mountParent(model);
+        const alignTool = fixture.querySelector('.o-tool[title="Horizontal align"]')!;
+        expect(alignTool.querySelector("svg")!.classList).toContain(expectedIconClass);
+        app.destroy();
+      }
+    );
   });
-  test.each([
-    ["text", {}, "align-left"],
-    ["0", {}, "align-right"],
-    ["0", { align: "left" }, "align-left"],
-    ["0", { align: "center" }, "align-center"],
-    ["0", { align: "right" }, "align-right"],
-  ])(
-    "alignment icon in top bar match the selected cell",
-    async (content, style, expectedIconClass) => {
+
+  describe("vertical align", () => {
+    test.each([
+      ["align-top", { verticalAlign: "top" }],
+      ["align-middle", { verticalAlign: "middle" }],
+      ["align-bottom", { verticalAlign: "bottom" }],
+    ])("can set vertical alignment with the toolbar", async (iconClass, expectedStyle) => {
       const model = new Model();
-      setCellContent(model, "A1", content);
-      model.dispatch("SET_FORMATTING", {
-        sheetId: model.getters.getActiveSheetId(),
-        target: [toZone("A1")],
-        style: style as Style,
-      });
+      selectCell(model, "A1");
       const { app } = await mountParent(model);
-      const alignTool = fixture.querySelector('.o-tool[title="Horizontal align"]')!;
-      expect(alignTool.querySelector("svg")!.classList).toContain(expectedIconClass);
+      const alignTool = fixture.querySelector('.o-tool[title="Vertical align"]')!;
+      alignTool.dispatchEvent(new Event("click"));
+      await nextTick();
+
+      const alignButtons = fixture.querySelectorAll("div.o-dropdown-item")!;
+      const button = [...alignButtons].find((element) =>
+        element.children[0]!.classList.contains(iconClass)
+      )!;
+
+      button.dispatchEvent(new Event("click"));
+      await nextTick();
+      expect(model.getters.getCurrentStyle()).toEqual(expectedStyle);
       app.destroy();
-    }
-  );
+    });
+    test.each([
+      ["text", {}, "align-middle"],
+      ["0", {}, "align-middle"],
+      ["0", { verticalAlign: "top" }, "align-top"],
+      ["0", { verticalAlign: "middle" }, "align-middle"],
+      ["0", { verticalAlign: "bottom" }, "align-bottom"],
+    ])(
+      "alignment icon in top bar match the selected cell",
```

Comment:
```suggestion
      "alignment icon in top bar matches the selected cell",
```
My english grammar isn't good, but I'm pretty sure that's more correct (ssame in the horizontal align test and wrapping test)